### ISSUE TITLE
"Add person" page not working in browsers other than Chrome

### ISSUE THREAD

**Issue by warpech** [StarcounterSamples/People#49]
### Steps to reproduce
1. Go to `http://localhost:8080/people/persons/add` in Firefox
2. Log in in SignIn
3. See that the workspace remains blank
4. Refresh the page
5. See that your are logged in, but the workspace remains blank
### Expected result

You should see the "add person" form.
### Actual result

Empty workspace appears.
##### Involved apps' version

Launcher 0f9268340de60fc88bb021a8387e9a5664ebe5a2 (current develop)
SignIn 397b233c8ac756c13f417bcb4bb9342f569e30ca (current master)
People efcd03b43839aeb93639da1424034cc67a134fb0 (current develop)
##### Browsers affected

- [ ] Chrome
- [x] Firefox
- [x] Edge
- [ ] Safari 9 (not tested)
- [ ] Safari 8 (not tested)
- [ ] IE 11 (not tested)
### Screenshot

![2016-10-28 17_09_31-launcher](https://cloud.githubusercontent.com/assets/566463/19811364/5640590e-9d31-11e6-8e4f-ada605e4a66a.png)

**Comment by warpech** [comment]
@ubbeK could you pls take a look?

**Comment by ubbeK** [comment]
@warpech Sure!

**Comment by ubbeK** [comment]
@warpech I am having trouble opening up starcounter-debug-aid in FF, tried two different computers, reinstallation and removed ad-block. It works fine for Chrome and Edge though. 

Does it work for you? Is this a sign to take weekend?

**Comment by ubbeK** [comment]
After some debugging the problem has been narrowed down to the [develop Launcher branch](https://github.com/StarcounterPrefabs/Launcher/tree/develop) since this issue does not exist when using Launcher master branch. More specifically, it is the starcounter-composition in PersonPage.html (People app) that does not load the content at all.

Note to self, another bug might be present, more debugging is required.
when debugging using Edge or FireFox, PersonPage.json.cs -> public void RefreshPerson(string ID = null) gets called with ID = %7B%7Bmodel.ImageUrl%7D%7D which translates into {{model.ImageUrl}}.
```
System.ArgumentException: ScErrBadArguments (SCERR1001): One or more arguments was invalid. Encoded string of ObjectID is of invalid size
Version: 2.2.2.3527.
Help page: https://github.com/Starcounter/Starcounter/wiki/SCERR1001.
   at Starcounter.DbHelper.Base64DecodeObjectID(String objectID) in C:\TeamCity\BuildAgent\work\sc-develop-nightly-544\Level1\src\Starcounter\Mock\DbHelper.cs:line 201
   at People.PersonPage.RefreshPerson(String ID) in C:\GitHub\Starcounter\People\src\People\ViewModels\PersonPage.json.cs:line 116
   at People.PartialHandlers.<>c__DisplayClass0_1.<Register>b__10() in C:\GitHub\Starcounter\People\src\People\Api\PartialHandlers.cs:line 84
   at Starcounter.Db.Scope[TResult](Func`1 func, Boolean isReadOnly) in C:\TeamCity\BuildAgent\work\sc-develop-nightly-544\Level1\src\Starcounter\Db.cs:line 484
   at People.PartialHandlers.<>c.<Register>b__0_9(String id) in C:\GitHub\Starcounter\People\src\People\Api\PartialHandlers.cs:line 79
   at lambda_method(Closure , Request , IntPtr , IntPtr )
   at Starcounter.Rest.UserHandlerInfo.RunUserDelegate(Request req, IntPtr methodSpaceUriSpaceOnStack, IntPtr parametersInfoOnStack) in C:\TeamCity\BuildAgent\work\sc-develop-nightly-544\Level1\src\Starcounter.Rest\UriHandlersManager.cs:line 221
   at Starcounter.Internal.Web.AppRestServer.RunDelegateAndProcessResponse(IntPtr methodSpaceUriSpaceOnStack, IntPtr parametersInfoOnStack, Request req) in C:\TeamCity\BuildAgent\work\sc-develop-nightly-544\Level1\src\Starcounter.Apps.JsonPatch\AppRestServer.cs:line 98
   at Starcounter.Rest.UriManagedHandlersCodegen.RunUriMatcherAndCallHandler(String methodSpaceUriSpace, String methodSpaceUriSpaceLower, Request req, UInt16 portNumber, Response& resp) in C:\TeamCity\BuildAgent\work\sc-develop-nightly-544\Level1\src\Starcounter.Rest\UriManagedHandlersCodegen.cs:line 1117
   at Starcounter.Self.DoSelfCall(UInt16 portNumber, String method, String relativeUri, Dictionary`2 headersDictionary, String body, Byte[] bodyBytes, HandlerOptions handlerOptions, Request req) in C:\TeamCity\BuildAgent\work\sc-develop-nightly-544\Level1\src\Starcounter.Internal\Rest\Self.cs:line 397
   at Starcounter.Self.GET(UInt16 port, String uri, Dictionary`2 headersDictionary, HandlerOptions ho) in C:\TeamCity\BuildAgent\work\sc-develop-nightly-544\Level1\src\Starcounter.Internal\Rest\Self.cs:line 133
   at Starcounter.UriMapping.<>c__DisplayClass18_0.<CallAllHandlersForClass>b__0(HandlerInfoForUriMapping x) in C:\TeamCity\BuildAgent\work\sc-develop-nightly-544\Level1\src\Starcounter\Apps\OntologyAndOrdinaryMapping.cs:line 833
   at Starcounter.UriMapping.CallAllHandlersForClass(List`1 resps, MappingClassInfo type, String stringParam, List`1 alreadyCalledHandlers) in C:\TeamCity\BuildAgent\work\sc-develop-nightly-544\Level1\src\Starcounter\Apps\OntologyAndOrdinaryMapping.cs:line 853
   at Starcounter.UriMapping.CallAllHandlersInTypeHierarchy(Request req, String className, String paramStr, Boolean alreadyHasResponse) in C:\TeamCity\BuildAgent\work\sc-develop-nightly-544\Level1\src\Starcounter\Apps\OntologyAndOrdinaryMapping.cs:line 737
   at Starcounter.UriMapping.ClassHierarchyCallProxy(Request req, String className, String paramStr) in C:\TeamCity\BuildAgent\work\sc-develop-nightly-544\Level1\src\Starcounter\Apps\OntologyAndOrdinaryMapping.cs:line 930
   at lambda_method(Closure , Request , IntPtr , IntPtr )
   at Starcounter.Rest.UserHandlerInfo.RunUserDelegate(Request req, IntPtr methodSpaceUriSpaceOnStack, IntPtr parametersInfoOnStack) in C:\TeamCity\BuildAgent\work\sc-develop-nightly-544\Level1\src\Starcounter.Rest\UriHandlersManager.cs:line 161
   at Starcounter.Internal.Web.AppRestServer.RunDelegateAndProcessResponse(IntPtr methodSpaceUriSpaceOnStack, IntPtr parametersInfoOnStack, Request req) in C:\TeamCity\BuildAgent\work\sc-develop-nightly-544\Level1\src\Starcounter.Apps.JsonPatch\AppRestServer.cs:line 98
   at Starcounter.Rest.UriManagedHandlersCodegen.RunUriMatcherAndCallHandler(String methodSpaceUriSpace, String methodSpaceUriSpaceLower, Request req, UInt16 portNumber, Response& resp) in C:\TeamCity\BuildAgent\work\sc-develop-nightly-544\Level1\src\Starcounter.Rest\UriManagedHandlersCodegen.cs:line 1117
   at Starcounter.Self.DoSelfCall(UInt16 portNumber, String method, String relativeUri, Dictionary`2 headersDictionary, String body, Byte[] bodyBytes, HandlerOptions handlerOptions, Request req) in C:\TeamCity\BuildAgent\work\sc-develop-nightly-544\Level1\src\Starcounter.Internal\Rest\Self.cs:line 397
   at Starcounter.Self.GET(UInt16 port, String uri, Dictionary`2 headersDictionary, HandlerOptions ho) in C:\TeamCity\BuildAgent\work\sc-develop-nightly-544\Level1\src\Starcounter.Internal\Rest\Self.cs:line 133
   at Starcounter.UriMapping.<>c__DisplayClass14_0.<OntologyMap>b__0(Request req, String appObjectId) in C:\TeamCity\BuildAgent\work\sc-develop-nightly-544\Level1\src\Starcounter\Apps\OntologyAndOrdinaryMapping.cs:line 604
   at lambda_method(Closure , Request , IntPtr , IntPtr )
   at Starcounter.Rest.UserHandlerInfo.RunUserDelegate(Request req, IntPtr methodSpaceUriSpaceOnStack, IntPtr parametersInfoOnStack) in C:\TeamCity\BuildAgent\work\sc-develop-nightly-544\Level1\src\Starcounter.Rest\UriHandlersManager.cs:line 161
   at Starcounter.Internal.Web.AppRestServer.RunDelegateAndProcessResponse(IntPtr methodSpaceUriSpaceOnStack, IntPtr parametersInfoOnStack, Request req) in C:\TeamCity\BuildAgent\work\sc-develop-nightly-544\Level1\src\Starcounter.Apps.JsonPatch\AppRestServer.cs:line 98
   at Starcounter.Rest.UriManagedHandlersCodegen.RunUriMatcherAndCallHandler(String methodSpaceUriSpace, String methodSpaceUriSpaceLower, Request req, UInt16 portNumber, Response& resp) in C:\TeamCity\BuildAgent\work\sc-develop-nightly-544\Level1\src\Starcounter.Rest\UriManagedHandlersCodegen.cs:line 1117
   at Starcounter.Self.DoSelfCall(UInt16 portNumber, String method, String relativeUri, Dictionary`2 headersDictionary, String body, Byte[] bodyBytes, HandlerOptions handlerOptions, Request req) in C:\TeamCity\BuildAgent\work\sc-develop-nightly-544\Level1\src\Starcounter.Internal\Rest\Self.cs:line 397
   at Starcounter.Self.GET[T](String uri, HandlerOptions ho) in C:\TeamCity\BuildAgent\work\sc-develop-nightly-544\Level1\src\Starcounter.Internal\Rest\Self.cs:line 22
   at People.MainHandlers.<>c__DisplayClass1_0.<GetLauncherPage>b__0() in C:\GitHub\Starcounter\People\src\People\Api\MainHandlers.cs:line 91
   at Starcounter.Db.Scope[TResult](Func`1 func, Boolean isReadOnly) in C:\TeamCity\BuildAgent\work\sc-develop-nightly-544\Level1\src\Starcounter\Db.cs:line 484
   at People.MainHandlers.GetLauncherPage(String Url, Boolean DbScope) in C:\GitHub\Starcounter\People\src\People\Api\MainHandlers.cs:line 90
   at People.MainHandlers.<>c.<Register>b__0_8(String id) in C:\GitHub\Starcounter\People\src\People\Api\MainHandlers.cs:line 71
   at People.AuthorizedHandle.<>c__DisplayClass1_1`1.<GET>b__1() in C:\GitHub\Starcounter\People\src\People\Helpers\AuthorizedHandle.cs:line 15
   at People.AuthorizedHandle.TrySignIn(Func`1 code) in C:\GitHub\Starcounter\People\src\People\Helpers\AuthorizedHandle.cs:line 21
   at People.AuthorizedHandle.<>c__DisplayClass1_0`1.<GET>b__0(T param) in C:\GitHub\Starcounter\People\src\People\Helpers\AuthorizedHandle.cs:line 15
   at lambda_method(Closure , Request , IntPtr , IntPtr )
   at Starcounter.Rest.UserHandlerInfo.RunUserDelegate(Request req, IntPtr methodSpaceUriSpaceOnStack, IntPtr parametersInfoOnStack) in C:\TeamCity\BuildAgent\work\sc-develop-nightly-544\Level1\src\Starcounter.Rest\UriHandlersManager.cs:line 221
   at Starcounter.Internal.Web.AppRestServer.RunDelegateAndProcessResponse(IntPtr methodSpaceUriSpaceOnStack, IntPtr parametersInfoOnStack, Request req) in C:\TeamCity\BuildAgent\work\sc-develop-nightly-544\Level1\src\Starcounter.Apps.JsonPatch\AppRestServer.cs:line 98
   at Starcounter.Internal.AppsBootstrapper.ProcessExternalRequest(Request req) in C:\TeamCity\BuildAgent\work\sc-develop-nightly-544\Level1\src\Starcounter.Apps.JsonPatch\AppsBootstrapper.cs:line 765
HResult=-2147024809
HelpLink=https://github.com/Starcounter/Starcounter/wiki/SCERR1001
```

**Comment by ubbeK** [comment]
Commit [Remove HTML Merger code as it is available in Starcounter](https://github.com/StarcounterPrefabs/Launcher/commit/37b9651b5ed83912d3b3ad59c4874c3a65cd29a9) seems to be the error for this issue. Any pointers on how to fix it @tomalec ?

**Comment by tomalec** [comment]
Find matching code in starcounter and check what is the difference.

**Comment by tomalec** [comment]
Or check if bindings works fine as requesting for {{ is deffinately a bug.

**Comment by warpech** [comment]
@ubbeK what is your version of Starcounter? Hopefully equivalent to the last Starcounter stable or newer?

**Comment by ubbeK** [comment]
@warpech It is definatly new than last stable. I think its from Friday, what number that was.

**Comment by tomalec** [comment]
After some investigation it seems that `juicy-composition`  is also involved. I'll continue investigation tomorrow evening.

**Comment by tomalec** [comment]
Note to self: So far it seems to be a polyfill issue, as imported document fragment when clone becomes empty in polyfilled browsers https://github.com/Starcounter/starcounter-include/blob/gh-pages/starcounter-include.html#L164

**Comment by tomalec** [comment]
I have managed to isolate it heavily. Remaining parts are: webcomponents.js, polymer, dom-bind, and just few lines from our code that contains:

- creating CE
- creating Shadow DOM
- dispatching an event

https://jsbin.com/raqili/2/edit?html,output

I suspect it's some kind of race condition between `dom-bind` stamping, Custom Element upgrade, and Shadow DOM polyfill.

I'll investigate it further and try to isolate even more.

The biggest doubt I have is that, most probably this is not an issue with V1, or we could fix it, by not-introducing during migration entire codebase from v0->v1 ;)

**Comment by tomalec** [comment]
I have managed to reduce this issue to polyfill issue https://github.com/webcomponents/webcomponentsjs/issues/648
However it's still quite tangled between Shadow DOM and MutationObserver polyfills

**Comment by tomalec** [comment]
It seems to work correctly with webcomponents.js#v1. @warpech Do you think we can wait until then?

**Comment by tomalec** [comment]
Now I cannot even try to find any workaround for WebComponents issue as 
- develop is not building
- the last commit that builds throws 500 for `/people/persons`
   ```log
Connection error: InternalServerError (500) 

GET http://localhost:8080/people/persons

Starcounter.DbException: ScErrSchemaCodeMismatch (SCERR4177): Operation failed be because input does not match schema. CLR Class is not loaded for table HelloWorld.Spender
Version: 2.2.2.3703.
Help page: https://github.com/Starcounter/Starcounter/wiki/SCERR4177.
   w Starcounter.Binding.Bindings.CreateExceptionOnTypeDefNotFound(Int32 tableId) w C:\TeamCity\TeamCity10\buildAgent\work\sc-develop-nightly-563\Level1\src\Starcounter\Binding\Bindings.cs:wiersz 460
   w Starcounter.Binding.Bindings.BuildTypeBindingFromTypeDef(Int32 tableId) w C:\TeamCity\TeamCity10\buildAgent\work\sc-develop-nightly-563\Level1\src\Starcounter\Binding\Bindings.cs:wiersz 332
   w Starcounter.Enumerator.MoveNext() w C:\TeamCity\TeamCity10\buildAgent\work\sc-develop-nightly-563\Level1\src\Starcounter\Mock\Enumerator.cs:wiersz 239
   w Starcounter.Query.Execution.IndexScan.MoveNextManaged() w C:\TeamCity\TeamCity10\buildAgent\work\sc-develop-nightly-563\Level1\src\Starcounter\Query\Execution\Enumerators\IndexScan.cs:wiersz 475
   w Starcounter.Query.Execution.IndexScan.MoveNext() w C:\TeamCity\TeamCity10\buildAgent\work\sc-develop-nightly-563\Level1\src\Starcounter\Query\Execution\Enumerators\IndexScan.cs:wiersz 550
   w Starcounter.Query.Execution.Sort.CreateEnumerator() w C:\TeamCity\TeamCity10\buildAgent\work\sc-develop-nightly-563\Level1\src\Starcounter\Query\Execution\Enumerators\Sort.cs:wiersz 115
   w Starcounter.Query.Execution.Sort.MoveNext() w C:\TeamCity\TeamCity10\buildAgent\work\sc-develop-nightly-563\Level1\src\Starcounter\Query\Execution\Enumerators\Sort.cs:wiersz 145
   w Starcounter.Json.Array_InitializeAfterImplicitConversion(Json parent, TObjArr template) w C:\TeamCity\TeamCity10\buildAgent\work\sc-develop-nightly-563\Level1\src\Starcounter.XSON\Class Json\Json.Array.cs:wiersz 85
   w Starcounter.Templates.TArray`1.BoundOrUnboundSet(Json parent, Arr`1 value) w C:\TeamCity\TeamCity10\buildAgent\work\sc-develop-nightly-563\Level1\src\Starcounter.XSON\Templates\TArray.cs:wiersz 119
   w People.PersonsPage.set_Persons(Arr`1 value) w D:\repos\PolyjuiceProjects\Polyjuice\People\src\People\obj\x64\Debug\ViewModels\ViewModels\PersonsPage.json:wiersz 44
   w People.PersonsPage.RefreshPersons() w D:\repos\PolyjuiceProjects\Polyjuice\People\src\People\ViewModels\PersonsPage.json.cs:wiersz 18
   w People.PartialHandlers.<>c.<Register>b__0_1() w D:\repos\PolyjuiceProjects\Polyjuice\People\src\People\Api\PartialHandlers.cs:wiersz 23
   w Starcounter.Rest.UserHandlerInfo.RunUserDelegate(Request req, IntPtr methodSpaceUriSpaceOnStack, IntPtr parametersInfoOnStack) w C:\TeamCity\TeamCity10\buildAgent\work\sc-develop-nightly-563\Level1\src\Starcounter.Rest\UriHandlersManager.cs:wiersz 221
   w Starcounter.Internal.Web.AppRestServer.RunDelegateAndProcessResponse(IntPtr methodSpaceUriSpaceOnStack, IntPtr parametersInfoOnStack, Request req) w C:\TeamCity\TeamCity10\buildAgent\work\sc-develop-nightly-563\Level1\src\Starcounter.Apps.JsonPatch\AppRestServer.cs:wiersz 98
   w Starcounter.Rest.UriManagedHandlersCodegen.RunUriMatcherAndCallHandler(String methodSpaceUriSpace, String methodSpaceUriSpaceLower, Request req, UInt16 portNumber, Response& resp) w C:\TeamCity\TeamCity10\buildAgent\work\sc-develop-nightly-563\Level1\src\Starcounter.Rest\UriManagedHandlersCodegen.cs:wiersz 1117
   w Starcounter.Self.DoSelfCall(UInt16 portNumber, String method, String relativeUri, Dictionary`2 headersDictionary, String body, Byte[] bodyBytes, HandlerOptions handlerOptions, Request req) w C:\TeamCity\TeamCity10\buildAgent\work\sc-develop-nightly-563\Level1\src\Starcounter.Internal\Rest\Self.cs:wiersz 397
   w Starcounter.Self.GET[T](String uri, HandlerOptions ho) w C:\TeamCity\TeamCity10\buildAgent\work\sc-develop-nightly-563\Level1\src\Starcounter.Internal\Rest\Self.cs:wiersz 22
   w People.MainHandlers.GetLauncherPage(String Url, Boolean DbScope) w D:\repos\PolyjuiceProjects\Polyjuice\People\src\People\Api\MainHandlers.cs:wiersz 94
   w People.MainHandlers.<>c.<Register>b__0_5() w D:\repos\PolyjuiceProjects\Polyjuice\People\src\People\Api\MainHandlers.cs:wiersz 53
   w People.AuthorizedHandle.<>c__DisplayClass0_0.<GET>b__1() w D:\repos\PolyjuiceProjects\Polyjuice\People\src\People\Helpers\AuthorizedHandle.cs:wiersz 9
   w People.AuthorizedHandle.TrySignIn(Func`1 code) w D:\repos\PolyjuiceProjects\Polyjuice\People\src\People\Helpers\AuthorizedHandle.cs:wiersz 21
   w People.AuthorizedHandle.<>c__DisplayClass0_0.<GET>b__0() w D:\repos\PolyjuiceProjects\Polyjuice\People\src\People\Helpers\AuthorizedHandle.cs:wiersz 9
   w Starcounter.Rest.UserHandlerInfo.RunUserDelegate(Request req, IntPtr methodSpaceUriSpaceOnStack, IntPtr parametersInfoOnStack) w C:\TeamCity\TeamCity10\buildAgent\work\sc-develop-nightly-563\Level1\src\Starcounter.Rest\UriHandlersManager.cs:wiersz 221
   w Starcounter.Internal.Web.AppRestServer.RunDelegateAndProcessResponse(IntPtr methodSpaceUriSpaceOnStack, IntPtr parametersInfoOnStack, Request req) w C:\TeamCity\TeamCity10\buildAgent\work\sc-develop-nightly-563\Level1\src\Starcounter.Apps.JsonPatch\AppRestServer.cs:wiersz 98
   w Starcounter.Internal.AppsBootstrapper.ProcessExternalRequest(Request req) w C:\TeamCity\TeamCity10\buildAgent\work\sc-develop-nightly-563\Level1\src\Starcounter.Apps.JsonPatch\AppsBootstrapper.cs:wiersz 765
ErrorCode=4177
HResult=-2146233088
HelpLink=https://github.com/Starcounter/Starcounter/wiki/SCERR4177

More information about this error may be available in the server error log.
```


As I already spend two weeks on this issee, I'm moving now to the others. Please, ping me once somebody will need any workaround for WC bug, I think I have few.

**Comment by warpech** [comment]
Can you please say what does it actually mean for us to go with webcomponents.js#v1. Is there a migration path that allows us to use it immediately, while keeping webcomponents.js#v0 and Polymer 1.7?

**Comment by tomalec** [comment]
Polymer team published migration path for Polymer elements and apps, but not for webcomponentsjs :( 
Let's hope to get some answer here https://twitter.com/tomalecpl/status/798204488246968321 or here
https://gitter.im/webcomponents/webcomponentsjs?at=5829e934df5ae9664556e75a

**Comment by tomalec** [comment]
As discussed with @warpech we agreed on plan as follows:

1. I'll try to find a workaround for this issue that could be implemented in `juicy-composition` so we could continue development as usual (time limit 24hrs),
2. I'll try to find a fix in webcomponentsjs, create a PR, and switch Starcounter to my own fork. Hope that they will merge it sooner than other PRs from May, or that upcoming version of webcomponentsjs is finally v1 (time limit 24hrs),
3. If tomorrow I will see no chance to do any of above, I'll investigate running v0 and v1 polyfills side by side. If it will work, it could hit the performance, especially on polyfiled and mobile browsers, but at least will give us opportunity to migrate to v1 sooner, easier and smoothly, (time limit 2days)
4. If all above will fail we will start migration all our stack to v1.

**Comment by tomalec** [comment]
Fixed in Starcounter 2.2.2.3750
